source/slang/slang-check-stmt.cpp

```diff
@@ -247,16 +288,17 @@ namespace Slang
         stmt->condition = CheckExpr(stmt->condition);
         subContext.checkStmt(stmt->body);
 
-        // TODO(tfoley): need to check that all case tags are unique
-
-        // TODO(tfoley): check that there is at most one `default` clause
+        // check the case value exits within the switch
+        validateCaseStmts(stmt, getSink());
     }
 
     void SemanticsStmtVisitor::visitCaseStmt(CaseStmt* stmt)
     {
-        // TODO(tfoley): Need to coerce to type being switch on,
-        // and ensure that value is a compile-time constant
         auto expr = CheckExpr(stmt->expr);
+
+        // coerce to type being switch on, and ensure that value is a compile-time constant
+        auto exprVal = CheckIntegerConstantExpression(expr, IntegerConstantExpressionCoercionType::AnyInteger, nullptr, ConstantFoldingKind::CompileTime);
```

Comment:
Should call `tryConstantFoldExpr` to allow enum typed IntVals.

Reply:
`CheckIntegerConstantExpression` does call `tryConstantFoldExpr` as well as attempting to coerce when we use `const int` values for cases. So, I feel that this is a better choice here.

Reply:
The difference is that CheckIntegetConstantExpr is trying to coerce the Val into an int after folding, which may fail if the enum type has an underlying type that is not int.

Reply:
Actually you are right. A falcor test with enum cases failed due to cost analysis:

```
C:\actions-runner\_work\slang\slang\FalcorBin\build\windows-vs2022\bin\Release\shaders\Scene/Scene.slang(662): error 30019: expected an expression of type 'int', got 'HitType'
```
The cost was >= 90000, and caused the folding to fail at this point at `slang-check-conversion.cpp`.
```
if (cost >= kConversionCost_Explicit)
```

I changed this to use `tryConstantFoldExpr`. And I'm not able to get the positive test case with int to pass, it fails with an internal error.
```
(0): error 99999: Slang compilation aborted due to an exception of class Slang::InternalError: assert failure: !inst || as<T>(inst)
```
I'm considering removing the extra test case.